### PR TITLE
Widget: Change text when expanding variations

### PR DESCRIPTION
**Pull request by ser8phin**
- [x] add changing text
- [x] use `computed-property`
- [x] add caret
- [x] style caret
- [x] fix naming

### CODE REVIEW

File: src/pretix/static/pretixpresale/scss/widget.scss [hunk 1]
```diff
@@ -429,6 +429,14 @@
...
   .pretix-widget-item-availability-col {
     text-align: center;
+
+    .pretix-widget-collapse-indicator-right:before {
+      content: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 4.233 4.233'%3E%3Cpath d='m0 0 2.1167 2.1167-2.1167 2.1167' /%3E%3C/svg%3E%0A");
+    }
+
+    .pretix-widget-collapse-indicator-down:before {
+      content: "<"
+    }
```
[the remark this answers is not in the document]
Comment (wiffbi) [reply src/pretix/static/pretixpresale/scss/widget.scss:439]: Amongst other things I already discussed that with ser8phin. This is definitely still a draft. :) Also should use the same icon as the shop IMHO.

[the remark this answers is not in the document]
Comment (ser8phin) [reply src/pretix/static/pretixpresale/scss/widget.scss]: Yes, that was planned, and both rotation, v-show and the svg are now adjusted. The color of the caret is still wrong though, working on it...

File: src/pretix/static/pretixpresale/js/widget/widget.js [hunk 2]
```diff
@@ -500,7 +501,9 @@ Vue.component('item', {
         // Availability
         + '<div class="pretix-widget-item-availability-col">'
         + '<a v-if="show_toggle" :href="\'#\' + item.id + \'-variants\'" @click.prevent.stop="expand" role="button" tabindex="0"'
-        + '   v-bind:aria-expanded="expanded ? \'true\': \'false\'" v-bind:aria-controls="item.id + \'-variants\'">'+ strings.variations + '</a>'
+        + '   v-bind:aria-expanded="expanded ? \'true\': \'false\'" v-bind:aria-controls="item.id + \'-variants\'">'
+        + '{{ variationsLink }}'
+        + '</a>'
```
Comment (wiffbi) [inline src/pretix/static/pretixpresale/js/widget/widget.js:506]: I would call it variationToggleLabel – Link sounds like a link or URL.

```suggestion
        + '   v-bind:aria-expanded="expanded ? \'true\': \'false\'" v-bind:aria-controls="item.id + \'-variants\'">{{ variationsToggleLabel }}</a>'
```

File: src/pretix/static/pretixpresale/js/widget/widget.js [hunk 3]
```diff
@@ -597,6 +600,9 @@ Vue.component('item', {
                 return this.$root.currency + " " + floatformat(this.item.min_price, 2);
             }
         },
+        variationsLink: function () {
```
Comment (wiffbi) [inline src/pretix/static/pretixpresale/js/widget/widget.js:603]:
```suggestion
        variationsToggleLabel: function () {
```

File: src/pretix/static/pretixpresale/scss/widget.scss [hunk 4]
```diff
@@ -429,8 +429,21 @@
...
+      height: $font-size-base;
+      background: url("data:image/svg+xml,%3Csvg viewBox='0 0 14 14' xmlns='http://www.w3.org/2000/svg' xml:space='preserve'%3E%3Cpath fill='#{url-friendly-colour($link-color)}' d='M6.395 4.151a.268.268 0 0 0-.177.077l-.386.386a.259.259 0 0 0-.077.177c.002.067.029.13.077.179l3.033 3.031-3.033 3.032a.255.255 0 0 0-.077.177.253.253 0 0 0 .077.178l.386.385a.268.268 0 0 0 .177.077.27.27 0 0 0 .178-.077l3.595-3.595a.259.259 0 0 0 .077-.177.255.255 0 0 0-.077-.176L6.573 4.228a.257.257 0 0 0-.178-.077Z'/%3E%3C/svg%3E");
+      transition: transform .5s;
+    }
+    a[aria-expanded=true]::before {
+      transform: rotate(90deg);
+    }
   }
 
+
```
Comment (wiffbi) [inline src/pretix/static/pretixpresale/scss/widget.scss:446]: Sorry, I added that empty line.

```suggestion
```

**Review by wiffbi — COMMENTED** [review]
Looks and works like a charm. I would change the name of that computed property if you are okay with it. Other than that perfect and ready to go!

File: src/pretix/static/pretixpresale/js/widget/widget.js [hunk 5]
```diff
@@ -51,6 +51,7 @@ var strings = {
     'close': django.pgettext('widget', 'Close'),
     'continue': django.pgettext('widget', 'Continue'),
     'variations': django.pgettext('widget', 'See variations'),
+    'hide_variations': django.pgettext('widget', 'Hide variations'),
```
Comment (wiffbi) [inline src/pretix/static/pretixpresale/js/widget/widget.js:54]: As discussed change to match pretix-presale:
```suggestion
    'variations': django.pgettext('widget', 'Show variants'),
    'hide_variations': django.pgettext('widget', 'Hide variants'),
```

**Review by wiffbi — APPROVED** [review]
Looks ready to me – so you can remove the draft status. I just added a comment regarding matching labels on showing variants with pretix-presale.

**Pull request merged**